doc/make.py

```diff
@@ -220,10 +220,11 @@ def html(self):
         if os.path.exists(zip_fname):
             os.remove(zip_fname)
 
-        if self.single_doc_html is not None:
-            self._open_browser(self.single_doc_html)
-        else:
-            self._add_redirects()
+        if ret_code == 0:
```

Comment:
do we already have a good error message to the user if the build fails?

Reply:
yes, we'll have the sphinx error message. In the example in the description, the end of the log will be:
```
Recursion error:
maximum recursion depth exceeded while calling a Python object
This can happen with very large or deeply nested source files.  You can carefully increase the default Python recursion limit of 1000 in conf.py with e.g.:
    import sys; sys.setrecursionlimit(1500)
```